app/services/claims/case_worker_claim_updater.rb

```diff
@@ -1,161 +1,64 @@
-      elsif @state == 'rejected'
-        add_error 'You cannot specify values when rejecting a claim'
-      end
-    end
-
-    def validate_state_when_no_value_params
-      return unless @state.in?(%w[authorised part_authorised])
-      add_error 'You must specify positive values if authorising or part authorising a claim'
-    end
-
-    def validate_reason_presence
-      return unless %w[rejected refused].include?(@state)
-      add_error("requires a reason when #{state_verb}", state_symbol(false)) if @transition_reasons&.empty?
-      add_error('needs a description', state_symbol) if transition_reason_text_missing?
-    end
-
-    def state_verb
-      @state_verb ||= @state.eql?('refused') ? 'refusing' : 'rejecting'
-    end
-
-    def state_symbol(other_suffix = true)
-      @state_noun ||= "#{@state}_reason#{'_other' if other_suffix}".to_sym
-    end
-
-    def transition_reason_text_missing?
-      @transition_reasons&.any? { |reason| %w[other other_refuse].include?(reason) } && @transition_reason_text.blank?
-    end
-
-    def params_present?(params)
-      return false unless params.present?
-      %w[fees expenses disbursements].any? { |field| params[field].to_f > 0.0 }
-    end
-
     def update_and_transition_state
-      event = Claims::InputEventMapper.input_event(@state)
+      event = Claims::InputEventMapper.input_event(state)
 
-      @claim.class.transaction do
-        @claim.update(@params)
-        update_assessment if @assessment_params_present
-        add_redetermination if @redetermination_params_present
-        @claim.send(event, audit_attributes) unless state_not_updateable?
-        add_message if @transition_reasons.present?
+      claim.class.transaction do
+        update_assessment if assessment_params
+        add_redetermination if redetermination_params
+        claim.send(event, audit_attributes) unless state_not_updateable?
+        add_message if transition_reasons.present?
       rescue StandardError => err
-        add_error err.message
+        add_error(err.message)
         raise ActiveRecord::Rollback
       end
     end
 
     def state_not_updateable?
-      @state.blank? || @state == @claim.state
+      state.blank? || state == claim.state
     end
 
     def update_assessment
-      params_with_defaults = {
-        'fees' => '0.00',
-        'expenses' => '0.00',
-        'disbursements' => '0.00'
-      }.merge(@assessment_params)
-      @claim.assessment.update(params_with_defaults)
+      claim.assessment.update(assessment_params)
     end
 
     def add_redetermination
-      params_with_defaults = {
-        'fees' => '0.00',
-        'expenses' => '0.00',
-        'disbursements' => '0.00'
-      }.merge(@redetermination_params)
-      @claim.redeterminations << Redetermination.new(params_with_defaults)
+      claim.redeterminations.create(redetermination_params)
     end
 
     # rubocop:disable Metrics/LineLength
     def add_message
```

Comment:
💭 To get rid of `# rubocop:disable Metrics/LineLength`:

```ruby
def add_message
  return unless Release.reject_refuse_messaging_released?
  claim.messages.create(sender_id: current_user.id, body: transition_message)
end
```

Reply:
👍